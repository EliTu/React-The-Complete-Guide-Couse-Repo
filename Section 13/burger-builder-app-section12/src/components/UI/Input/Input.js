import React from 'react';
import styles from './Input.module.css';
import PropTypes from 'prop-types';

const Input = props => {
	// props:
	const { elementType, elementConfig, value, handleChange } = props;
	// CSS Modules styles:
	const { Input } = styles;

	let inputElement = null;

	switch (elementType) {
		case 'input':
			inputElement = (
				<input
					{...elementConfig}
					value={value}
					onChange={handleChange}
				/>
			);
			break;

		case 'textarea':
			inputElement = (
				<textarea
					{...elementConfig}
					value={value}
					onChange={handleChange}
				/>
			);
			break;

		case 'select':
			inputElement = (
				<select value={value} onChange={handleChange}>
					{elementConfig.options.map(option => (
						<option value={option.value} key={option.value}>
							{option.displayValue}
						</option>
					))}
				</select>
			);
			break;

		default:
			inputElement = <input {...elementConfig} value={value} />;
	}
	return (
		<div className={Input}>
			<label>{elementConfig.label}</label>
			{inputElement}
		</div>
	);
};

Input.propTypes = {
	elementType: PropTypes.string,
	elementConfig: PropTypes.object,
};

export default Input;
