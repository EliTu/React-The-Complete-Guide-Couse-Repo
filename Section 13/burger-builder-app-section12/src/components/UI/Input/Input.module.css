.Input {
    width: 100%;
    padding: 0.4rem;
    box-sizing: border-box;
    font-family: inherit;
}

.Input input,
.Input textarea,
.Input select {
    background-color: rgba(241, 238, 238, 0.993);
    color: #181616e0;
    font-size: 1rem;
    width: 40%;
    outline: none;
    border: none;
    border-bottom: 2px solid rgba(105, 241, 105, 0.808);
    margin: 1rem auto;
    display: block;
    text-align: center;
    transition: all 0.5s;
}

/* To disable up and down number spinners on number input field */
.Input input[type=number]::-webkit-inner-spin-button,
.Input input[type=number]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
}

.Input textarea {
    border: 2px solid rgba(105, 241, 105, 0.808);
    overflow: auto;
    resize: none;
    text-align: start;
    margin: 1rem auto;
    height: 50%;
    width: 75%;
}

.Input label {
    font-weight: 700;
    display: block;
    margin-bottom: 1rem;
}

.Input select option {
    text-align: center;
    background-color: rgba(241, 238, 238, 0.993);
    color: #181616e0;
}

.Input input:focus {
    width: 55%;
    border-radius: 7px;
    border-bottom: 2px solid rgba(3, 136, 3, 0.781);
}

.Input select:focus,
.Input textarea:focus {
    border-bottom: 2px solid rgba(3, 136, 3, 0.781);
}

.Input textarea:focus {
    border: 2px solid rgba(3, 136, 3, 0.781);
}

.Input .InvalidStyle {
    border-bottom: 2px solid rgba(190, 16, 16, 0.781);
}