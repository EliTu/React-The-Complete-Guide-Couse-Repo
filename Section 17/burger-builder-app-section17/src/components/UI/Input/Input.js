import React from 'react';
import styles from './Input.module.css';
import PropTypes from 'prop-types';

const Input = props => {
	// props:
	const {
		elementType,
		elementConfig,
		value,
		handleChange,
		isInvalid,
		errorMessage,
		hasUserInput,
	} = props;

	// CSS Modules styles:
	const { Input, InvalidStyle, errorMessageStyle } = styles;

	let inputElement = null;
	let errorMessageElement = (
		<p className={errorMessageStyle}>{errorMessage}</p>
	);

	switch (elementType) {
		case 'input':
			inputElement = (
				<input
					className={isInvalid && hasUserInput ? InvalidStyle : null}
					{...elementConfig}
					value={value}
					onChange={handleChange}
				/>
			);
			break;

		case 'textarea':
			inputElement = (
				<textarea
					{...elementConfig}
					value={value}
					onChange={handleChange}
				/>
			);
			break;

		case 'select':
			inputElement = (
				<select value={value} onChange={handleChange}>
					{elementConfig.options.map(option => (
						<option value={option.value} key={option.value}>
							{option.displayValue}
						</option>
					))}
				</select>
			);
			break;

		default:
			inputElement = (
				<input
					{...elementConfig}
					value={value}
					className={isInvalid ? InvalidStyle : null}
				/>
			);
	}
	return (
		<div className={Input}>
			<label>{elementConfig.label}</label>
			{inputElement}
			{isInvalid && hasUserInput ? errorMessageElement : null}
		</div>
	);
};

Input.propTypes = {
	elementType: PropTypes.string,
	elementConfig: PropTypes.object,
	isInvalid: PropTypes.bool,
	errorMessage: PropTypes.string,
	hasUserInput: PropTypes.bool,
	handleChange: PropTypes.func,
};

export default Input;
