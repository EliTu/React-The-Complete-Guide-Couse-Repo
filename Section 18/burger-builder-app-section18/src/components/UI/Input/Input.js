import React, { useRef, useEffect } from 'react';
import styles from './Input.module.css';
import PropTypes from 'prop-types';

const Input = props => {
	// props:
	const {
		elementType,
		elementConfig,
		value,
		validation,
		focused,
		handleChange,
		handleEnterPress,
	} = props;

	// CSS Modules styles:
	const { Input, InvalidStyle, errorMessageStyle } = styles;

	let inputElement = null;
	let errorMessageElement = (
		<p className={errorMessageStyle}>{validation.errorMessage}</p>
	);

	// Listen to keyboard enter click to submit form:
	const enterPressCallback = (event, func) => {
		if (event.key === 'Enter') func(event);
	};

	// Focus the first input field upon component mount
	const focusRef = useRef();
	useEffect(() => {
		if (focused) focusRef.current.focus();
	}, [focused]);

	switch (elementType) {
		case 'input':
			inputElement = (
				<input
					ref={focusRef}
					className={
						!validation.valid && validation.hasUserInput
							? InvalidStyle
							: null
					}
					{...elementConfig}
					value={value}
					onChange={handleChange}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
				/>
			);
			break;

		case 'textarea':
			inputElement = (
				<textarea
					{...elementConfig}
					value={value}
					onChange={handleChange}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
				/>
			);
			break;

		case 'select':
			inputElement = (
				<select value={value} onChange={handleChange}>
					{elementConfig.options.map(option => (
						<option value={option.value} key={option.value}>
							{option.displayValue}
						</option>
					))}
				</select>
			);
			break;

		default:
			inputElement = (
				<input
					{...elementConfig}
					value={value}
					className={!validation.valid ? InvalidStyle : null}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
				/>
			);
	}
	return (
		<div className={Input}>
			<label>{elementConfig.label}</label>
			{inputElement}
			{!validation.valid && validation.hasUserInput
				? errorMessageElement
				: null}
		</div>
	);
};

Input.propTypes = {
	elementType: PropTypes.string,
	elementConfig: PropTypes.object,
	validation: PropTypes.object,
	focused: PropTypes.bool,
	handleChange: PropTypes.func,
	handleEnterPress: PropTypes.func,
};

export default Input;
