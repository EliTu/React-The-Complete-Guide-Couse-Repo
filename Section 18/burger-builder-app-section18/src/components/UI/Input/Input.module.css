.Input {
    width: 100%;
    padding: 0.4rem;
    box-sizing: border-box;
    font-family: inherit;
}

.Input input,
.Input textarea,
.Input select {
    background-color: rgba(241, 238, 238, 0.993);
    color: #181616e0;
    font-size: 0.9rem;
    width: 40%;
    outline: none;
    border: none;
    border-bottom: 2px solid rgba(105, 241, 105, 0.808);
    margin: 1rem auto;
    display: block;
    text-align: center;
    transition: all 0.5s;
}

/* To disable up and down number spinners on number input field */
.Input input[type=number]::-webkit-inner-spin-button,
.Input input[type=number]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
}

.Input textarea {
    border: 1px solid rgba(105, 241, 105, 0.808);
    overflow: auto;
    resize: none;
    text-align: start;
    margin: 1rem auto;
    height: 50%;
    width: 75%;
}

.Input label {
    font-weight: 700;
    display: block;
    margin-bottom: 1rem;
}

.Input select option {
    text-align: center;
    background-color: rgba(241, 238, 238, 0.993);
    color: #181616e0;
}

.Input input:focus {
    width: 55%;
}

.Input .InvalidStyle {
    border-bottom: 2px solid rgba(190, 16, 16, 0.63);
}

.errorMessageStyle {
    font-size: 0.8rem;
    color: rgba(190, 16, 16, 0.788);
}

::placeholder {
    color: rgba(58, 50, 50, 0.438);
}