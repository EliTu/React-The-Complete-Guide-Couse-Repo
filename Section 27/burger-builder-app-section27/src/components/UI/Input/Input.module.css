@import '../../../variables.css';

.Input {
    width: 100%;
    padding: 0.4rem;
    box-sizing: border-box;
    font-family: inherit;
}

.Input input,
.Input textarea,
.Input select {
    background-color: rgba(241, 238, 238, 0);
    color: var(--coloring-dark);
    font-size: 0.9rem;
    width: 55%;
    outline: none;
    border: none;
    border-bottom: 2px solid var(--coloring-opaque-dark);
    margin: 1rem auto;
    display: block;
    text-align: center;
    transition: all 0.5s;
}

/* To disable up and down number spinners on number input field */
.Input input[type=number]::-webkit-inner-spin-button,
.Input input[type=number]::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
}

.Input textarea {
    border: 1px solid var(--coloring-opaque-dark);
    overflow: auto;
    resize: none;
    text-align: start;
    margin: 1rem auto;
    height: 50%;
    width: 75%;
}

.Input label {
    font-weight: 700;
    display: block;
    margin-bottom: 1rem;
}

.Input select option {
    text-align: center;
    background-color: rgba(241, 238, 238, 0.993);
    color: var(--coloring-dark);
}

.Input input:focus {
    width: 70%;
}

.Input .ValidStyle {
    border-bottom: 2px solid var(--glow-green-opaque-lighter);
}

.Input .InvalidStyle {
    border-bottom: 2px solid var(--invalid-red-dark);
}

.errorMessageStyle {
    font-size: 0.8rem;
    color: var(--invalid-red-light);
}

::placeholder {
    color: var(--coloring-opaque-dark);
}

@media screen and (max-width: 800px) {
    .Input input {
        width: 60%;
    }
}