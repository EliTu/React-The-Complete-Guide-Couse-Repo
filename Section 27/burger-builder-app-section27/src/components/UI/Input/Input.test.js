import React from 'react';
import Input from './Input';
import { shallow } from 'enzyme';
import { findByTestAttr } from '../../../utilities/test-utilities/findByTestAttr';

const setComponentProps = (elemType, configType, value, isValid) => {
	const props = {
		elementType: `${elemType}`,
		elementConfig: {
			type: `${configType}`,
			placeholder: 'abc',
			label: 'abc',
		},
		value: `${value}`,
		validation: {
			required: true,
			hasUserInput: false,
			valid: `${isValid}`,
			errorMessage: 'abc',
		},
		focused: true,
		handleChange: () => [],
		handleEnterPress: () => [],
		data: 'abc',
	};
	return props;
};

const setComponent = (props = {}) => {
	const component = shallow(<Input {...props} />);
	return component;
};

describe('<Input/>', () => {
	describe('Component (not testing the input fields)', () => {
		let props = setComponentProps('input', 'text', 'abc', true);
		let component = setComponent(props);
		expect(component).toBeTruthy();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	describe('Type: input="text" test', () => {
		it('should render without errors', () => {
			let props = setComponentProps('input', 'text', 'abc', true);
			let component = setComponent(props);
			let input = findByTestAttr(component, 'input-test');
			expect(input.length).toBe(1);
		});
		
		it('should not render if elementType is not input ', () => {
			let props = setComponentProps('password', 'text', 'abc', true);
			let component = setComponent(props);
			let input = findByTestAttr(component, 'input-test');
			expect(input.length).toBe(0);
		});
	});
});
