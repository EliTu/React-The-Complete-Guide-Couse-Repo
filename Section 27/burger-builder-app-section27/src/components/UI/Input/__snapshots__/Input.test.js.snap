// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Component (not testing the input fields) should render an Input component withot errors 1`] = `
<div
  className="Input"
>
  <label>
    abc
  </label>
  <input
    className=""
    data-test="input-test"
    label="abc"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="abc"
    type="text"
    value="abc"
  />
</div>
`;

exports[`<Input/> Type: input="text" test should not render if elementType is not input  1`] = `
<div
  className="Input"
>
  <label>
    abc
  </label>
  <input
    className={null}
    label="abc"
    onKeyPress={[Function]}
    placeholder="abc"
    type="text"
    value="abc"
  />
</div>
`;

exports[`<Input/> Type: input="text" test should render without errors 1`] = `
<div
  className="Input"
>
  <label>
    abc
  </label>
  <input
    className=""
    data-test="input-test"
    label="abc"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="abc"
    type="text"
    value="abc"
  />
</div>
`;

exports[`<Input/> Type: input="textarea" test should render without errors 1`] = `
<div
  className="Input"
>
  <label>
    abc
  </label>
  <textarea
    data-test="textarea-test"
    label="abc"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="abc"
    type=""
    value="abc"
  />
</div>
`;
